pkg/sentry/control/proc.go

```diff
@@ -404,3 +475,20 @@ func ContainerUsage(kr *kernel.Kernel) map[string]uint64 {
 	}
 	return cusage
 }
+
+// createFDMap creates the file descriptor map from the unmarshalled ExecArgs.
+func (args *ExecArgs) createFDMap() (map[int]*fd.FD, error) {
+	files := args.GetFiles()
```

Comment:
I think we still need to validate that `len(guestFDs) == len(files)` and return an error if they are different.  Otherwise this code could panic.

Reply:
This is one reason why I liked the original complexity. As there was no possibility to create a `FilePayload` that violated the constraint by enforcing the use of the helper methods, this code could only have panicked if one had manually sent a malformed RPC payload.